apps/credential-showcase-api-server/src/database/repositories/PersonaRepository.ts

```diff
@@ -14,47 +14,79 @@ class PersonaRepository implements RepositoryDefinition<Persona, NewPersona> {
   ) {}
 
   async create(newPersona: NewPersona): Promise<Persona> {
-    if (newPersona.headshotImage) {
-      const headshotImage = await this.assetRepository.findById(newPersona.headshotImage)
-      if (!headshotImage) {
-        throw new NotFoundError(`Headshot image not found for persona ${newPersona.id}`)
+    if (newPersona.headshotImage !== undefined && newPersona.headshotImage !== null) {
+      try {
+        await this.assetRepository.findById(newPersona.headshotImage)
+      } catch (error) {
+        throw new NotFoundError(`No asset found for id: ${newPersona.headshotImage}`)
       }
     }
 
-    if (newPersona.bodyImage) {
-      const bodyImage = await this.assetRepository.findById(newPersona.bodyImage)
-      if (!bodyImage) {
-        throw new NotFoundError(`Body image not found for persona ${newPersona.id}`)
+    if (newPersona.bodyImage !== undefined && newPersona.bodyImage !== null) {
+      try {
+        await this.assetRepository.findById(newPersona.bodyImage)
+      } catch (error) {
+        throw new NotFoundError(`No asset found for id: ${newPersona.bodyImage}`)
```

Comment:
not needed as 

await this.assetRepository.findById(newPersona.bodyImage)

is laready throwing this error

Reply:
also not the responsibility of this repo to manage this error

Reply:
just call assetRepository.findById twice, 1 time for bodyImage and 1 time for headshotImage